HopsanGUI/Widgets/TextEditorWidget.h

```diff
@@ -35,50 +35,98 @@
 #define SCRIPTEDITOR_H
 
 #include <QWidget>
-#include <QTextEdit>
+#include <QPlainTextEdit>
 #include <QFileInfo>
 #include <QCompleter>
 
 #include "common.h"
+#include "Utilities/HighlightingUtilities.h"
 
-class HcomEditor : public QTextEdit
+
+class TextEditor : public QPlainTextEdit
 {
     Q_OBJECT
 public:
-    HcomEditor(QWidget *parent = 0);
+    TextEditor(HighlighterTypeEnum language, QWidget *parent = 0);
+
+    void lineNumberAreaPaintEvent(QPaintEvent *pEvent);
+    int lineNumberAreaWidth();
 
 protected:
+    void resizeEvent(QResizeEvent *pEvent);
+    void insertFromMimeData(const QMimeData *pData);
     void keyPressEvent(QKeyEvent *event);
+    void wheelEvent(QWheelEvent* e);
 
 public slots:
     void updateAutoCompleteList();
 
 private slots:
+    void updateLineNumberAreaWidth(int newBlockCount);
+    void highlightCurrentLine();
+    void updateLineNumberArea(const QRect &, int);
     void insertCompletion(const QString& completion);
 
 private:
+    HighlighterTypeEnum mLanguage;
+    QWidget *mpLineNumberArea;
     QCompleter *mpCompleter;
 };
 
-class ScriptEditor : public QWidget
+
+class LineNumberArea : public QWidget
+{
+public:
+    LineNumberArea(TextEditor *editor) : QWidget(editor)
+    {
+        mpTextEditor = editor;
+    }
+
+    QSize sizeHint() const
+    {
+        return QSize(mpTextEditor->lineNumberAreaWidth(), 0);
+    }
+
+protected:
+    void paintEvent(QPaintEvent *event)
+    {
+        mpTextEditor->lineNumberAreaPaintEvent(event);
+    }
+
+private:
+    TextEditor *mpTextEditor;
+};
+
+
+class TextEditorWidget : public QWidget
 {
     Q_OBJECT
 public:
-    explicit ScriptEditor(QFileInfo file, QWidget *parent = nullptr);
+    explicit TextEditorWidget(QFileInfo file, HighlighterTypeEnum highlighter, QWidget *parent = nullptr);
 
-    QFileInfo getScriptFileInfo() const { return mScriptFileInfo; }
+    QFileInfo getFileInfo() const { return mFileInfo; }
     bool isSaved() const { return mIsSaved; }
 
+public slots:
+    void  find(QString text, QTextDocument::FindFlags flags);
+
 protected:
     void wheelEvent(QWheelEvent* event);
 
 signals:
 
 private:
-    HcomEditor *mpEditor;
-    QFileInfo mScriptFileInfo;
+    void setHighlighter(HighlighterTypeEnum highlighter);
+
+    TextEditor *mpEditor;
+    QFileInfo mFileInfo;
     bool mIsSaved = true;
     QString mSavedText;
+    HcomHighlighter *mpHcomHighlighter;
```

Comment:
These should probably be std::unique_ptr see other comment

Reply:
I added deleteLater() in constructor, as mentioned above.